src/server/RedirectingHttpHandler.ts

```diff
@@ -0,0 +1,106 @@
+import type { TargetExtractor } from '../http/input/identifier/TargetExtractor';
+import { RedirectResponseDescription } from '../http/output/response/RedirectResponseDescription';
+import type { ResponseWriter } from '../http/output/ResponseWriter';
+import { getLoggerFor } from '../logging/LogUtil';
+import { FoundHttpError } from '../util/errors/FoundHttpError';
+import { MovedPermanentlyHttpError } from '../util/errors/MovedPermanentlyHttpError';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import { PermanentRedirectHttpError } from '../util/errors/PermanentRedirectHttpError';
+import type { RedirectHttpError } from '../util/errors/RedirectHttpError';
+import { SeeOtherHttpError } from '../util/errors/SeeOtherHttpError';
+import { TemporaryRedirectHttpError } from '../util/errors/TemporaryRedirectHttpError';
+import { getRelativeUrl, joinUrl } from '../util/PathUtil';
+import type { HttpHandlerInput } from './HttpHandler';
+import { HttpHandler } from './HttpHandler';
+import type { HttpRequest } from './HttpRequest';
+
+const redirectErrorFactories: Record<301 | 302 | 303 | 307 | 308, (location: string) => RedirectHttpError> = {
+  301: (location: string): RedirectHttpError => new MovedPermanentlyHttpError(location),
+  302: (location: string): RedirectHttpError => new FoundHttpError(location),
+  303: (location: string): RedirectHttpError => new SeeOtherHttpError(location),
+  307: (location: string): RedirectHttpError => new TemporaryRedirectHttpError(location),
+  308: (location: string): RedirectHttpError => new PermanentRedirectHttpError(location),
+};
```

Comment:
You can also do `new RedirectHttpError(statusCode, location, '');` to dynamically generate the correct error. The named errors are mostly used to hide the exact status codes a bit in the code so it's clearer what exactly happens, but that's not necessary here since that's what we get as input.

The new errors can stay though. 4.0.0 does completely change how we create new error classes which may cause issues when doing the rebase to 4.0.0 so I can look into that if there's an issue.

Reply:
I tried that first, but `RedirectHttpError` is an abstract class.

Reply:
Ah woops. It's not in the 4.0.0 branch 😅 . But fine to leave it like this then.